src/protocol/boolean/bitwise_gt_constant.rs

```diff
@@ -0,0 +1,249 @@
+use super::or::or;
+use crate::error::Error;
+use crate::ff::Field;
+use crate::protocol::{context::Context, BitOpStep, RecordId};
+use crate::secret_sharing::Arithmetic as ArithmeticSecretSharing;
+
+/// Compares the `[a]` and `c`, and returns `1` iff `a > c`
+///
+/// Rabbit: Efficient Comparison for Secure Multi-Party Computation
+/// 2.1 Comparison with Bitwise Shared Input – `LTBits` Protocol
+/// Eleftheria Makri, et al.
+/// <https://eprint.iacr.org/2021/119.pdf>
+///
+/// ## Errors
+/// Lots of things may go wrong here, from timeouts to bad output. They will be signalled
+/// back via the error response
+pub async fn bitwise_greater_than_constant<F, C, S>(
+    ctx: C,
+    record_id: RecordId,
+    a: &[S],
+    c: u32,
+) -> Result<S, Error>
+where
+    F: Field,
+    C: Context<F, Share = S>,
+    S: ArithmeticSecretSharing<F>,
+{
+    let first_diff_bit = first_differing_bit(&ctx, record_id, a, c).await?;
+
+    // Compute the dot-product [a] x `first_diff_bit`. 1 iff a > c.
+    // We can swap `a` with `c` to yield 1 iff a < c. We just need to convert
+    // `c` to `&[c]` using `local_secret_shared_bits`.
+    ctx.narrow(&Step::DotProduct)
+        .sum_of_products(record_id, &first_diff_bit, a)
+        .await
+}
+
+async fn first_differing_bit<F, C, S>(
+    ctx: &C,
+    record_id: RecordId,
+    a: &[S],
+    b: u32,
```

Comment:
If you're going to use a u32 for `b`, then you should `assert` that `a.len() <= 32`. Or use a u128, and assert `<= 128`